### PR TITLE
enable local css source maps with flag

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
This is an attempt to prevent the `npm watch` task from crashing after multiple re-builds are triggered. The issue was tracked down to `css-loader` having source maps on by default. Memory profiles suggest that webpack or the css-loader is retaining old build's strings in memory. 

I believe [this](https://github.com/webpack-contrib/css-loader/issues/886) is the core issue and will revisit adding css source maps to the default watch task when it's resolved. 

This change turns off css sourcemaps by default for local development. A new script has been added that 
- turns on css source maps
- increases the memory node is allowed to use

The new script will probably still crash after node runs out of memory but this will give it more room to fail and limit that failure to work that requires debugging css with sourcemaps. 

## Testing done
`npm run watch:css-sourcemaps` after 20+ rebuilds: 
screen shot 2019-01-18 at 13 31 46

`npm run watch` after 20+ rebuilds:
screen shot 2019-01-18 at 13 35 49

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jbalboni — APPROVED** [review]
This seems reasonable to me, curious to see what @ncksllvn thinks.

File: config/webpack.config.js [hunk 1]
```diff
@@ -43,6 +43,10 @@ const configGenerator = (buildOptions, apps) => {
     ENVIRONMENTS.VAGOVPROD,
   ].includes(buildOptions.buildtype);
 
+  const enableCSSSourcemaps =
+    buildOptions.buildtype !== ENVIRONMENTS.LOCALHOST ||
+    buildOptions['local-css-sourcemaps'];
```
Comment (ncksllvn) [inline config/webpack.config.js]: If `buildOptions['entry']` is passed, should it flip to `true`? Just thinking that if we've limited what apps we're compiling, then maybe it should consider CSS sourcemaps to be safe too.

File: README.md [hunk 2]
```diff
@@ -24,7 +24,8 @@ very secret.
...
 | build the site with optimizitons (minification, chunking etc) on. | Set `NODE_ENV=production` before running build. |
 | reset local environment (clean out node modules and runs npm install) | `npm run reset:env`                      |
-| run the site for local development with automatic rebuilding of Javascript and sass | `npm run watch` then visit `http://localhost:3001/`. You may also set `buildtype` and `NODE_ENV` though setting `NODE_ENV` to production will make incremental builds slow. |
+| run the site for local development with automatic rebuilding of Javascript and sass without css sourcemaps | `npm run watch` then visit `http://localhost:3001/`. You may also set `buildtype` and `NODE_ENV` though setting `NODE_ENV` to production will make incremental builds slow. |
+| run the site for local development with automatic rebuilding of Javascript and sass with css sourcemaps| `npm run watch:css-sourcemaps` then visit `http://localhost:3001/`. You may also set `buildtype` and `NODE_ENV` though setting `NODE_ENV` to production will make incremental builds slow. |
```
Comment (ncksllvn) [inline README.md]: It might be worth adding a note about the CSS source maps being heavy on memory and that's why we separate them out.

**Review by ncksllvn — APPROVED** [review]
LGTM, although I wonder if there's a [setting](https://webpack.js.org/plugins/source-map-dev-tool-plugin/) or different type of source map that could help free up some memory.  It might be worth experimenting with this [line](https://github.com/department-of-veterans-affairs/vets-website/pull/9415/files#diff-a58d55bdb5770c78ad512f8e91f8d051R224) sometime and see if it makes any difference.

Sometime, I wonder if we could implement a `help` command or something as a way of documenting these args. This seems like one that feels natural as just as an extra arg passed to the regular watch task instead of a separate way of starting the site, but without it in the `package.json` it's hard to make it visible.

**Pull request merged**